### PR TITLE
Fix index projects tests

### PR DESCRIPTION
**Pull request by ibilal**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -39,6 +39,8 @@ app.use((req, res, next) => {
 });
 
 app.use(express.static(path.join(__dirname, 'public')));
+
+app.use(express.static(path.join(__dirname, 'public')));
```
Comment (chances) [inline app.js]: Remove this duplicate, preserving whitespace between routes

**Comment by ibilal** [comment]
Do not merge until the teamName environment variable branch has been merged into development. There will be some conflicts with the tests and it will be better to fix before merging in.

File: test/routes/index.js [hunk 2]
```diff
@@ -10,106 +10,237 @@ let chaiPromise = require('chai-as-promised');
...
+    });
+    it('should render the index view with an empty list of projects when authenticates promise is resolved', (done) => {
+      resolvedPromiseProxySetup();
+      validateReturnFlag = true;
+      chai.request(app)
+        .post('/')
+        .send(userLoginMockData)
+        .then(res => {
+          expect(res).to.have.status(200);
+          expect(res).to.redirect.to('/projects');
```
Comment (RickyV33) [inline test/routes/index.js]: This should redirect to the graph endpoint

File: test/routes/index.js [hunk 3]
```diff
@@ -10,106 +10,237 @@ let chaiPromise = require('chai-as-promised');
...
+};
+
+/**
+ * This function sets up the proxies required to test the index route in the event the get request cannot be found
+ * in the source file. It sets up the app to proxyquire app.js with the stubbed index. The index stub is set up to
+ * proxyquire the index route with a fake index route that is missing the get request source.
+ *
+ */
+function indexNotFoundProxySetup () {
+  index = require('./mockIndexRoute');
```
Comment (RickyV33) [inline test/routes/index.js]: This is only used once and it doesn't really need its own file. You should be able to include this here. Maybe something like:
```let router = express.Router();```
Then stub out `./routes/index: router`

Or something like that...

File: test/routes/projects.js [hunk 4]
```diff
@@ -5,35 +5,50 @@
...
+          expect(res).to.have.status(200);
+          expect(res).to.have.property('text');
+          expect(res.text).to.equal(renderedView);
+        });
+    });
+    it('should render projects view with projects when session.projects is not null', function () {
+      data = {title: 'Projects', projects: [{id: 1, name: 'Project 1'}, {id: 2, name: 'Project 2'}]};
+      path = join(__dirname, '../../views/partials/projects.ejs');
+      renderedView = ejs.compile(read(path, 'utf8'), {filename: path})(data);
+      app = require('./mockApp');
```
Comment (RickyV33) [inline test/routes/projects.js]: you should be able to just require the app here and then add a new method to it

```
let app = require(...)
app.use((req, res, next) => {
req.session.projects = [{id: 1, name: 'Project 1'}, {id: 2, name: 'Project 2'}];
next();
});```

**Comment by RickyV33** [comment]
LGTM other than those comments I left about the extra files

**Pull request merged**